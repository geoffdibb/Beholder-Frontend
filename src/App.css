.App {
  text-align: center;
  margin: auto;
  margin-left: auto;
  margin-right: auto;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 40vmin;
  pointer-events: none;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}
colours{
  color:#EED98D;
  color:#E8D1C0;
  color:#DAE1DD;
  color:#ADB3A4;
  color:#474C55;

}

.App-link {
  color: #61dafb;
}

.Logo{
  margin: 40px;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.Nav {
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}
