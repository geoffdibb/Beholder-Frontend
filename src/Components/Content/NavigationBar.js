import React from 'react';
import {
    Collapse,
    Navbar,
    NavbarToggler,
    NavbarBrand,
    Nav,
    NavItem,
    NavLink,
    Col
} from 'reactstrap';
import BackButton from './BackButton';


export default class NavigationBar extends React.Component {
    constructor(props) {
        super(props);

        this.toggle = this.toggle.bind(this);
        this.state = {
            isOpen: false
        };
    }
    toggle() {
        this.setState({
            isOpen: !this.state.isOpen
        });
    }
    render() {
        return (
            <div>
                <Navbar color="dark" dark expand="md">
                    <Col md='4'>
                        <Nav className="ml-auto" navbar>
                            <NavItem>
                                <NavLink href="/" >Home</NavLink>
                            </NavItem>
                            <NavItem>
                                <BackButton />
                            </NavItem>
                                    
                            <NavItem>
                                <NavLink href="/Results">Search Results</NavLink>
                            </NavItem>
                        </Nav>
                    </Col>
                    <Col md='4'>
                        <NavbarBrand className="mx-auto" href="/"  >
                            <img
                                src="../logo.png"
                                width="160"
                                height="39"
                                alt="Beholder logo"
                            />
                        </NavbarBrand>
                    </Col>
                    <Col md='4'>
                        <NavbarToggler onClick={this.toggle} />
                        <Collapse isOpen={this.state.isOpen} navbar>
                            <Nav className="ml-auto" navbar>
                                <NavItem>
                                    <NavLink href="/" >Log Out</NavLink>
                                </NavItem>
                            </Nav>
                        </Collapse>
                    </Col>
                </Navbar>
            </div>
        );
    }
}