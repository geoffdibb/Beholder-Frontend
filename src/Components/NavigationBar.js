import React from 'react';
import {
    Collapse,
    Navbar,
    NavbarToggler,
    NavbarBrand,
    Nav,
    NavItem,
    NavLink
} from 'reactstrap';

export default class NavigationBar extends React.Component {
    constructor(props) {
        super(props);

        this.toggle = this.toggle.bind(this);
        this.state = {
            isOpen: false
        };
    }
    toggle() {
        this.setState({
            isOpen: !this.state.isOpen
        });
    }
    render() {
        return (
            <div>
                <Navbar color="dark" dark expand="md">
                    <NavbarBrand href="/">Beholder</NavbarBrand>
                    <NavbarToggler onClick={this.toggle} />
                    <Collapse isOpen={this.state.isOpen} navbar>
                        <Nav className="ml-auto" navbar>
                            <NavItem>
                                <NavLink href="./Register">tab</NavLink>
                            </NavItem>
                            <NavItem>
                                <NavLink href="./Login">tab</NavLink>
                            </NavItem>
                            <NavItem>
                                <NavLink href="./UserList">tab</NavLink>
                            </NavItem>
                        </Nav>
                    </Collapse>
                </Navbar>
            </div>
        );
    }
}