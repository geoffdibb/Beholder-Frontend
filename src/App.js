import React from 'react';
import { BrowserRouter as Router, Route } from "react-router-dom";
import Keycloak from 'keycloak-js';

import './App.css';
import Homepage from './Components/Content/Homepage';
import Login from './Components/Login';

import { KeycloakProvider } from 'react-keycloak';


  const keycloak = new Keycloak();

const keycloakProviderInitConfig = {
  onLoad: 'check-sso',
};

export default class App extends React.Component {



  render() {
    return (
      <div className='App'>
        <Router>
          <div>

            <Route path="/Home" component={Homepage} />

            <Route exact path="/" component={Login} />

            {/* <Route path="/MapPage" component={MapPage} />

          <Route path="/Profile" component={Profile} />

          <Route path="/ResultPage" component={ResultPage} />

          <Route path="/SearchPage" component={SearchPage} />

          <Route path="/Audit" component={Audit} /> */}
          </div>
        </Router>
      </div>
    )
  }
}