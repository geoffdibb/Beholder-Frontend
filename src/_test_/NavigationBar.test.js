import React from 'react';
import { ReactDOM } from 'react-dom';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import {RouterComponent} from '../App';
import NavigationBar from '../Components/Content/NavigationBar';

Enzyme.configure({ adapter: new Adapter() })


it('snapshot navigation', () => {

    const tree = renderer.create(
    <MemoryRouter>
    <NavigationBar/>
  </MemoryRouter>
  ).toJSON()
    expect(tree).toMatchSnapshot();
  })
  
//   it('renders NavBar component', () => {
//     const wrapper = shallow(<RouterComponent />);
//     const content = <NavigationBar />;
//     expect(wrapper.contains(content)).toEqual(true);
//   });