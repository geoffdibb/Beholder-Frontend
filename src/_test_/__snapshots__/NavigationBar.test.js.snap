// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot navigation 1`] = `
<div>
  <nav
    className="navbar navbar-expand-md navbar-dark bg-dark"
  >
    <div
      className="col-md-4"
    >
      <ul
        className="ml-auto navbar-nav"
      >
        <li
          className="nav-item"
        >
          <a
            aria-current="page"
            className="nav-link active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            Back
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link"
            href="/Results"
            onClick={[Function]}
          >
            Search Results
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link"
            href="/Audit"
            onClick={[Function]}
          >
            Audit
          </a>
        </li>
      </ul>
    </div>
    <div
      className="col-md-4"
    >
      <a
        aria-current="page"
        className="mx-auto navbar-brand active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        <img
          alt="Beholder logo"
          height="39"
          src="../logo.png"
          width="160"
        />
      </a>
    </div>
    <div
      className="col-md-4"
    >
      <button
        className="navbar-toggler"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        style={Object {}}
      >
        <ul
          className="ml-auto navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/"
              onClick={[Function]}
            >
              Log Out
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;
